tests/system_tests/test_core/test_wandb_run.py

```diff
@@ -97,6 +98,56 @@ def test_init_with_settings(user, test_settings):
         assert run.settings.x_disable_stats
 
 
+def test_init_with_api_key_validation():
+    """Test that init raises AuthenticationError when given an invalid API key."""
+    # System tests setup already provides a valid API key in env var WANDB_API_KEY
+    # The env var gets parsed automatically to settings.api_key.
+    # Run settings is merged from global settings and init settings.
+    # Passing invalid key in init settings to trigger error so we can verify
+    # the key for login is from the merged run settings instead of global settings.
+    api_key = "invalid-api-key"
+    with pytest.raises(AuthenticationError, match=r"API key must have 40\+ characters"):
+        wandb.init(settings=wandb.Settings(api_key=api_key))
+
+
+def test_init_with_api_key_no_netrc(user, tmp_path):
+    """Test when init get api key from settings, it will not read or write netrc file."""
+    netrc_path = str(tmp_path / "netrc")
+    # Reset the valid API key from system tests setup
+    with mock.patch.dict(
+        os.environ,
+        {"NETRC": netrc_path, "WANDB_API_KEY": "invalid-api-key"},
+    ):
+        # No netrc before init
+        assert not os.path.exists(netrc_path)
+
+        # Use the valid key for init settings
+        with wandb.init(settings=wandb.Settings(api_key=user)) as run:
+            assert run is not None
+            assert run.id is not None
+
+        # No netrc after init
+        assert not os.path.exists(netrc_path)
+
+
+def test_init_with_api_key_public_api(user):
+    """Test run._public_api() uses the api key from init settings and caches the instance."""
+    # Reset the valid API key from system tests setup
+    with mock.patch.dict(os.environ, {"WANDB_API_KEY": "invalid-api-key"}):
+        # Use the valid key for init settings
+        with wandb.init(settings=wandb.Settings(api_key=user)) as run:
```

Comment:
This isn't a robust test because it relies on how the `autouse` fixtures are set up. It might start passing without testing anything if `conftest.py` is modified.

This is another one where you might as well use a mock and assert that the `public.Api` constructor is called a certain number of times (once) and with the correct arguments (an API key). I think you already have a unit test that does this, so this test is not needed.